semantics/src/main/kotlin/com/strumenta/kolasu/semantics/symbol/provider/SymbolProvider.kt

```diff
@@ -0,0 +1,8 @@
+package com.strumenta.kolasu.semantics.symbol.provider
+
+import com.strumenta.kolasu.model.Node
+import com.strumenta.kolasu.semantics.symbol.description.SymbolDescription
+
+interface SymbolProvider {
+    fun symbolFor(node: Node): SymbolDescription
```

Comment:
I would expect the symbol provider _not_ to produce symbols for all nodes that are not visible outside the AST

Reply:
Correct, will fix it